### PR TITLE
[Bots][Addedpythonsection][1411558]

### PR DESCRIPTION
**Pull request by v-rodriguezd**
Added python sections to bot basics and bot logic sections. Please review.

### CODE REVIEW

**Review by KirtiPereira — COMMENTED** [review]
Hi @laujan - please review / merge the PR.
Thank you.

**Review by KirtiPereira — CHANGES_REQUESTED** [review]
@v-rodriguezd: Please work on the comments.
In the Teams activity handler class, there are two primary Teams activity handle
KP: missing period.

The *Teams* activity handler will first check for any Teams specific events and pass it along to the Bot Framework's activity handler if none are found.
KP: Rephrase the sentence.

In the Teams activity handler class, there are two primary Teams activity handlers, `on_conversation_update_activity` that routes all conversation update activities and `on_invoke_activity` that routes all Teams invoke activities.
KP: Rephrase the sentence.

For each of these handlers, there is no base implementation, so just add the logic that you want in your override.
KP: Rephrase the sentence.

The invoke activities listed above are for conversational bots in Teams. 
KP: Replace ‘above’ with a link or an appropriate word as per the style guide.

KP: Please check the commas in the ‘Description’ column of the Teams-specific handlers section. 

KP: Please correct the capitalization in the ‘Description’ column of the Teams invoke activities section.

File: msteams-platform/bots/bot-basics.md [hunk 1]
```diff
@@ -135,6 +143,56 @@ Here is a list of all of the Teams activity handlers called from the `onInvokeAc
 | task/fetch                      | `handleTeamsTaskModuleFetch`        | Teams Task Module Fetch. |
 | task/submit                     | `handleTeamsTaskModuleSubmit`       | Teams Task Module Submit. |
 
-The invoke activities listed above are for conversational bots in Teams. The Bot Framework SDK also supports invocations specific to messaging extensions. For more information see [What are messaging extensions](https://aka.ms/azure-bot-what-are-messaging-extensions)
+The invoke activities listed above are for conversational bots in Teams. The Bot Framework SDK also supports invocations specific to messaging extensions. For more information, see [What are messaging extensions](https://aka.ms/azure-bot-what-are-messaging-extensions).
+
```
Comment (KirtiPereira) [inline msteams-platform/bots/bot-basics.md:147]: Replace 'above' with a link or an appropriate term as per the Style guide.

File: msteams-platform/bots/bot-basics.md [hunk 2]
```diff
@@ -135,6 +143,56 @@ Here is a list of all of the Teams activity handlers called from the `onInvokeAc
...
+| teamRenamed | `on_teams_team_renamed` | `return Task.CompletedTask;` Override this to handle a Teams Team being Renamed. For more information see [Team renamed](https://aka.ms/azure-bot-subscribe-to-conversation-events#team-renamed) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
+| MembersAdded | `on_teams_members_added` | Calls the `OnMembersAddedAsync` method in `ActivityHandler`. Override this to handle members joining a team. For more information see [Team member added](https://aka.ms/azure-bot-subscribe-to-conversation-events#Team-Member-Added) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
+| MembersRemoved | `on_teams_members_removed` | Calls the `OnMembersRemovedAsync` method in `ActivityHandler`. Override this to handle members leaving a team. For more information see [Team member removed](https://aka.ms/azure-bot-subscribe-to-conversation-events#Team-Member-Removed) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
+
```
Comment (KirtiPereira) [inline msteams-platform/bots/bot-basics.md:180]: Please check the commas in the ‘Description’ column of the table.

File: msteams-platform/bots/bot-basics.md [hunk 3]
```diff
@@ -135,6 +143,56 @@ Here is a list of all of the Teams activity handlers called from the `onInvokeAc
...
+| :-----------------------------  | :----------------------------------- | :----------------------------------------------------------- |
+| CardAction.Invoke               | `on_teams_card_action_invoke`       | Teams Card Action Invoke. |
+| fileConsent/invoke              | `on_teams_file_consent_accept`      | Teams File Consent Accept. |
+| fileConsent/invoke              | `on_teams_file_consent`            | Teams File Consent. |
+| fileConsent/invoke              | `on_teams_file_consent_decline`     | Teams File Consent. |
+| actionableMessage/executeAction | `on_teams_o365_connector_card_action` | Teams O365 Connector Card Action. |
+| signin/verifyState              | `on_teams_signin_verify_state`      | Teams Sign in Verify State. |
+| task/fetch                      | `on_teams_task_module_fetch`        | Teams Task Module Fetch. |
+| task/submit                     | `on_teams_task_module_submit`       | Teams Task Module Submit. |
+
```
Comment (KirtiPereira) [inline msteams-platform/bots/bot-basics.md:195]: Please correct the capitalization in the ‘Description’ column of the table.

**Review by KirtiPereira — CHANGES_REQUESTED** [review]

**Comment by v-rodriguezd** [comment]
> @v-rodriguezd: Please work on the comments.
> In the Teams activity handler class, there are two primary Teams activity handle
> KP: missing 's' and period.
> 
> The _Teams_ activity handler will first check for any Teams specific events and pass it along to the Bot Framework's activity handler if none are found.
> KP: Rephrase the sentence.
> 
> In the Teams activity handler class, there are two primary Teams activity handlers, `on_conversation_update_activity` that routes all conversation update activities and `on_invoke_activity` that routes all Teams invoke activities.
> KP: Rephrase the sentence.
> 
> For each of these handlers, there is no base implementation, so just add the logic that you want in your override.
> KP: Rephrase the sentence.
> 
> The invoke activities listed above are for conversational bots in Teams.
> KP: Replace ‘above’ with a link or an appropriate word as per the style guide.
> 
> KP: Please check the commas in the ‘Description’ column of the Teams-specific handlers section.
> 
> KP: Please correct the capitalization in the ‘Description’ column of the Teams invoke activities section.

Hi Kirti I have updated the document with the changes. Please review. Thank you.

File: msteams-platform/bots/bot-basics.md [hunk 4]
```diff
@@ -104,37 +112,87 @@ The handlers defined in `ActivityHandler` are outlined below.
...
+The invoke activities listed in the Teams invoke activities section are for conversational bots in Teams. The Bot Framework SDK also supports invoke activities specific to messaging extensions. For more information, see [What are messaging extensions](https://aka.ms/azure-bot-what-are-messaging-extensions).
+
+# [Python](#tab/python)
+
+#### Core Bot Framework handlers
+
+All of the activity handlers described in this section will continue to work as they do with a non-Teams bot, except for the *added* and *removed* members' activities. Activity handlers will be different in the context of a team, where the new member is added to the team as opposed to a message thread.
+
+Here is a list of the handlers defined in `ActivityHandler`:
```
Comment (surbhigupta12) [inline msteams-platform/bots/bot-basics.md:154]: Maintain consistency using "Here" and "Follows"

File: msteams-platform/bots/bot-basics.md [hunk 5]
```diff
@@ -104,37 +112,87 @@ The handlers defined in `ActivityHandler` are outlined below.
...
+| teamRenamed | `on_teams_team_renamed` | `return Task.CompletedTask;` Override this to handle a Teams team being renamed. For more information, see [Team renamed](https://aka.ms/azure-bot-subscribe-to-conversation-events#team-renamed) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
+| MembersAdded | `on_teams_members_added` | Calls the `OnMembersAddedAsync` method in `ActivityHandler`. Override this to handle members joining a team. For more information, see [Team member added](https://aka.ms/azure-bot-subscribe-to-conversation-events#team-members-added) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
+| MembersRemoved | `on_teams_members_removed` | Calls the `OnMembersRemovedAsync` method in `ActivityHandler`. Override this to handle members leaving a team. For more information, see [Team member removed](https://aka.ms/azure-bot-subscribe-to-conversation-events#team-members-removed) in [Conversation update events](https://aka.ms/azure-bot-subscribe-to-conversation-events).|
```
Comment (surbhigupta12) [inline msteams-platform/bots/bot-basics.md:179]: Please fix parallelism issues and rephrase where applicable.

**Review by surbhigupta12 — CHANGES_REQUESTED** [review]
File reviewed. Please update.

**Review by KirtiPereira — APPROVED** [review]
@surbhigupta12 - please approve and merge.
Thank you.

**Review by surbhigupta12 — APPROVED** [review]
Approving this PR.

**Pull request merged**